### PR TITLE
Create email event receipts when sending hearings notifications

### LINKED ISSUE #13751: Store information related to virtual hearing emails
### Description
To display information about virtual hearing emails to Veterans and their PoAs, and to triage in bat team, write a tech spec for backend changes in order to create new table to:
- Store email type, email subject line, and/or preview
- The initial hearing time included in the sent email

### Acceptance criteria
- [x] write a tech spec for backend changes ... #13907
- [x] implement schema changes ... #14005
- [ ] save email events to database ... #14027
- [ ] Check data is successfully saved in production

### Background/context/resources
This work has been broken out from Display more info about sent Virtual Hearings emails (#13370), which was based on usability testing with Hearing Coordinators (#12960)

### PR DESCRIPTION
**Pull request by ferristseng**
Resolves #13751 

Blocked by: 
  - [x] Schema Change in #14005 
  - [x] Upgrade to `rspec` in #14011

### Description

  * Create `SendHearingEmailEvent` model
  * Update email sending to create a new historical event record
  * Fix some issues with testing jobs.

    I discovered that `retry_on` code is never actually called when using `#perform_now` -- you have to use `#perform_later`. I ended up using some newer testing features in an updated version of `rspec` to test with `#perform_later`, hence the blocker on #14011. 
    * Fix code inside of `retry_on`
  * Updated `CreateConferenceJob` to throw an exception if either: the conference wasn't created, or emails weren't sent. The exception triggers a retry of the job.
  * Update `SendEmail` to attempt sending emails even if a previous email failed to send (e.g. if the veteran email failed to send, it will still attempt to send the representative email and judge email)

### Database Changes

* [x] Timestamps (created_at, updated_at) for new tables
* [x] Column comments updated
* [x] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [x] DB schema docs updated with `make docs`
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: app/models/hearings/sent_hearing_email_event.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  # Email types are specified in `SendEmail#email_for_recipient`
+  enum email_type: (
+    {
+      confirmation: "confirmation",
+      cancellation: "cancellation",
+      updated_time_confirmation: "updated_time_confirmation"
+    }
+  ), _prefix: :is
+
+  def readonly?
```
Comment (sahalliburton) [inline app/models/hearings/sent_hearing_email_event.rb:22]: @tomas-nava is this needed for this record? I am not seeing it being used anywhere else so I just want to be sure, thanks!

File: db/schema.rb [hunk 2]
```diff
@@ -1377,13 +1377,15 @@
     t.boolean "representative_email_sent", default: false, null: false, comment: "Whether or not a notification email was sent to the veteran's representative"
     t.boolean "request_cancelled", default: false, comment: "Determines whether the user has cancelled the virtual hearing request"
     t.datetime "updated_at", null: false
+    t.bigint "updated_by_id", comment: "The ID of the user who most recently updated the virtual hearing"
```
Comment (sahalliburton) [inline db/schema.rb:1380]: @tomas-nava this column seems to be named differently from the migration file, is this something rails is adding for us?

File: app/models/hearings/sent_hearing_email_event.rb (hunk 1 above)
Comment (tomas-nava) [reply app/models/hearings/sent_hearing_email_event.rb:22]: This is letting Rails know that we want this to be a read only record; if you try to update it after it's saved, it'll error. [Documentation here.](https://apidock.com/rails/ActiveRecord/Base/readonly%3f)

File: db/schema.rb (hunk 2 above)
Comment (tomas-nava) [reply db/schema.rb:1380]: Yes, we're using shorthand in the migration file to establish a relation and Rails implements it this way.

File: app/models/hearings/sent_hearing_email_event.rb (hunk 1 above)
Comment (sahalliburton) [reply app/models/hearings/sent_hearing_email_event.rb:22]: Thats really good to know thanks!

File: db/schema.rb (hunk 2 above)
Comment (sahalliburton) [reply db/schema.rb:1380]: Cool thats what I thought, but wanted to make sure I understood what was happening thanks!

**Pull request merged**